bazel/format/clang_tidy/clang_tidy.bzl

```diff
@@ -79,23 +60,25 @@ def _run_tidy(
     )
     return outfile
 
-def _rule_sources(ctx):
-    def check_valid_file_type(src):
-        """
-        Returns True if the file type matches one of the permitted srcs file types for C and C++ header/source files.
-        """
-        permitted_file_types = [
-            ".c", ".cc", ".cpp", ".cxx", ".c++", ".C", ".h", ".hh", ".hpp", ".hxx", ".inc", ".inl", ".H",
-        ]
-        for file_type in permitted_file_types:
-            if src.basename.endswith(file_type):
-                return True
-        return False
 
+def _check_valid_file_type(src):
+    """
+    Returns True if the file type matches one of the permitted srcs file types for C and C++ header/source files.
+    """
+    permitted_file_types = [
+        ".c", ".cc", ".cpp", ".cxx", ".c++", ".C", ".h", ".hh", ".hpp", ".hxx", ".inc", ".inl", ".H",
+    ]
+    for file_type in permitted_file_types:
+        if src.basename.endswith(file_type):
+            return True
```

Comment:
can we use dict here? for loop with more than 10 items seems a bit unnecessary linear search

Reply:
not sure i follow exactly - in real python there are various ways to optimize this but i dont think they work in starlark - can you give example?

Reply:
```

PERMITTED_TYPES = {"c": True, "cpp": True}

def perimtted(file_name):
    if file_name.split(".")[-1] in PERMITTED_TYPES:
        return True
```

like this?

Reply:
not sure if the dict would be faster than a list

what i think we can do is ...

```suggestion
    permitted_file_types = [
        ".c", ".cc", ".cpp", ".cxx", ".c++", ".C", ".h", ".hh", ".hpp", ".hxx", ".inc", ".inl", ".H",
    ]
    return src.extension in permitted_file_types
```

Reply:
probably - we can shift the list out and get rid of the fun also

Reply:
at least can you move `.h` and `.cc` to the first elements?

Reply:
they are the most common in envoy code base

Reply:
re `.h` i think we dont want it to run over the headers directly it causes a stack of other issues

if we _do_ want that then we need to fish them out from the `hdrs` attr